syntax = "proto3";

import "google/protobuf/timestamp.proto";

import "beams/service/remote_calls/generic_message.proto";
import "beams/service/remote_calls/behavior_tree.proto";

/*
  Message to be returned upon "heartbeat" request to program from client
  Informs client about application state in general
*/

message HeartBeatReply {
  MessageType mess_t = 1;
  // when our program is ticking many trees this may become a repeated or mapped fields
  repeated BehaviorTreeUpdateMessage behavior_tree_update = 2;
  google.protobuf.Timestamp reply_timestamp = 3;
  // maybe program uptime and git hash??
}
